source/elements/oneMath/source/architecture/execution_model.inc.rst

```diff
@@ -0,0 +1,91 @@
+
+Device Usage
+++++++++++++
+
+oneMath itself does not currently provide any interfaces for controlling device usage: for instance, controlling the number of cores used on the CPU, or the number of execution units on a GPU. However, such functionality may be available by partitioning a ``sycl::device`` instance into subdevices, when supported by the device.
+
+When given a queue associated with such a subdevice, a oneMath implementation shall only perform computation on that subdevice.
+
+.. _onemath_asynchronous_synchronous_execution:
+
+Asynchronous Execution
+++++++++++++++++++++++
+The oneMath API is designed to allow asynchronous execution of computational routines, to facilitate concurrent usage of multiple devices in the system. Each computational routine enqueues work to be performed on the selected device, and may (but is not required to) return before execution completes.
+
+Hence, it is the calling application's responsibility to ensure that any inputs are valid until computation is complete, and likewise to wait for computation completion before reading any outputs. This can be done automatically when using DPC++ buffers, or manually when using Unified Shared Memory (USM) pointers, as described in the sections below.
+
+Unless otherwise specified, asynchronous execution is *allowed*, but not *guaranteed*, by any oneMath computational routine, and may vary between implementations and/or versions. oneMath implementations must clearly document whether execution is guaranteed to be asynchronous for each supported routine. Regardless, calling applications shall not launch any oneMath computational routine with a dependency on a future oneMath API call, even if this computational routine executes asynchronously (i.e. a oneMath implementation may assume no antidependencies are present). This guarantee allows oneMath implementations to reserve resources for execution without risking deadlock.
+
+.. _onemath_synchronization_with_buffers:
+
+Synchronization When Using Buffers
+***********************************
+
+``sycl::buffer`` objects automatically manage synchronization between kernel launches linked by a data dependency (either read-after-write, write-after-write, or write-after-read).
+
+oneMath routines are not required to perform any additional synchronization of ``sycl::buffer`` arguments.
+
+.. _onemath_synchronization_with_usm:
+
+Synchronization When Using USM APIs
+***********************************
+
+When USM pointers are used as input to, or output from, a oneMath routine, it becomes the calling application's responsibility to manage possible asynchronicity.
+
+To help the calling application, all oneMath routines with at least one USM pointer argument also take an optional reference to a list of *input events*, of type ``std::vector<sycl::event>``, and have a return value of type ``sycl::event`` representing computation completion::
+
+    sycl::event math::domain::routine(..., std::vector<sycl::event> &in_events = {});
+
```

Comment:
we need to add `const` to the front of these types: `const std::vector<sycl::event>&`

Reply:
Agree, that looks like a small fix that can be done as part of this PR

Reply:
yeah, I hesitated to toss it into here, but is small enough to just make the change ...